pkg/dynamicinstrumentation/diconfig/location_expression.go

```diff
@@ -0,0 +1,322 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package diconfig
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ditypes"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/exp/rand"
```

Comment:
I think you want `math/rand` or `math/rand/v2` here

Reply:
math/rand API was deprecated for crypto/rand, using that here.